### PR TITLE
AOT Test

### PR DESCRIPTION
**Pull request by sindhuvahinis**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: .github/workflows/llm_integration.yml [hunk 1]
```diff
@@ -327,10 +327,65 @@ jobs:
...
+      - name: Download models and dockers
+        working-directory: serving/docker
+        run: |
+          docker pull deepjavalibrary/djl-serving:$DJLSERVING_DOCKER_TAG
+      - name: Test opt-6.7b
+        working-directory: tests/integration
+        run: |
+          rm -rf models
+          python3 llm/prepare.py deepspeed_aot opt-6.7b
+          docker run -t --rm --network=host \
```
Comment (lanking520) [inline .github/workflows/llm_integration.yml:359]: could we make this part of the launch script?

File: tests/integration/llm/client.py [hunk 2]
```diff
@@ -229,6 +237,26 @@ def test_ds_raw_model(model):
                 assert float(memory) / 1024.0 < spec["max_memory_per_gpu"]
 
 
+def test_ds_aot_handler(model):
```
Comment (lanking520) [inline tests/integration/llm/client.py:240]: what is special for AOT Handler? Could we just use DS raw handler?

File: tests/integration/llm/deepspeed-model.py [hunk 3]
```diff
@@ -15,7 +15,16 @@ def load_model(properties):
...
     logging.info(f"Loading model in {model_location}")
-    model = AutoModelForCausalLM.from_pretrained(model_location,
+    checkpoint = None
+    if "checkpoint" in properties:
+        checkpoint = os.path.join(model_location, properties['checkpoint'])
+
+    if checkpoint:
+        config_file = os.path.join(model_location, "config.json")
+        config = AutoConfig.from_pretrained(config_file)
+        model = AutoModelForCausalLM.from_config(config)
```
Comment (lanking520) [inline tests/integration/llm/deepspeed-model.py:25]: You may also want to use meta tensor loading?

File: .github/workflows/llm_integration.yml (hunk 1 above)
Comment (sindhuvahinis) [reply .github/workflows/llm_integration.yml:359]: Okay

File: .github/workflows/llm_integration.yml [hunk 4]
```diff
@@ -327,10 +327,65 @@ jobs:
...
+          yes | docker system prune -a --volumes
+          sudo rm -rf /home/ubuntu/actions-runner/_work/_tool/Java_Corretto_jdk/
+          echo "wait dpkg lock..."
+          while sudo fuser /var/{lib/{dpkg,apt/lists},cache/apt/archives}/lock >/dev/null 2>&1; do sleep 5; done
+      - name: Set up Python3
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10.x'
+      - name: Install pip dependencies
+        run: pip3 install requests pillow
```
Comment (siddvenk) [inline .github/workflows/llm_integration.yml:347]: why is pillow required?

File: .github/workflows/llm_integration.yml [hunk 5]
```diff
@@ -327,10 +327,61 @@ jobs:
...
+      - name: Download models and dockers
+        working-directory: serving/docker
+        run: |
+          docker pull deepjavalibrary/djl-serving:$DJLSERVING_DOCKER_TAG
+      - name: Test opt-6.7b
+        working-directory: tests/integration
+        run: |
+          rm -rf models
+          python3 llm/prepare.py deepspeed_aot opt-6.7b
+          ./launch_container.sh deepjavalibrary/djl-serving:$DJLSERVING_DOCKER_TAG $PWD/models deepspeed \
```
Comment (lanking520) [inline .github/workflows/llm_integration.yml:359]: let's define partition as a standalone step in Action. And check if file is there at the end. And loading the partition as a different step.

File: tests/integration/llm/client.py (hunk 2 above)
Comment (sindhuvahinis) [reply tests/integration/llm/client.py]: Re-using DS raw handler now

File: tests/integration/llm/deepspeed-model.py (hunk 3 above)
Comment (sindhuvahinis) [reply tests/integration/llm/deepspeed-model.py]: Added it and added checkpoint and base_dir to deepspeed.init_inference

File: .github/workflows/llm_integration.yml (hunk 4 above)
Comment (sindhuvahinis) [reply .github/workflows/llm_integration.yml]: No not needed. Removed it.

File: .github/workflows/llm_integration.yml (hunk 5 above)
Comment (sindhuvahinis) [reply .github/workflows/llm_integration.yml:359]: Okay, changed it into two steps.

**Comment by sindhuvahinis** [comment]
https://github.com/deepjavalibrary/djl-serving/actions/runs/4239523069/jobs/7367675614 - Test cases passed here

**Review by lanking520 — APPROVED** [review]

**Pull request merged**